buildsystem/modules/FindPython.cmake

```diff
@@ -43,6 +43,29 @@ function(py_get_config_var VAR RESULTVAR)
 	set("${RESULTVAR}" "${RESULT}" PARENT_SCOPE)
 endfunction()
 
+function(py_get_lib_name RESULTVAR)
+	# uses py_exec to compute Python's C/C++ library name, just like python-config does.
+	py_get_config_var(VERSION PYTHON_VERSION)
+	if(NOT "${PYTHON_VERSION}" VERSION_LESS "3.2")
+		set(USE_ABIFLAGS "use")
```

Comment:
@TheJJ , currently we are creating a list of all the python interpreters and try to compile a python extension using each interpreter. The version check for python is in the form of a static assert in `FindPython_test.cpp`.